tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -13288,6 +13322,8 @@ void hlsl::HandleDeclAttributeForHLSL(Sema &S, Decl *D, const AttributeList &A,
             "intersection,anyhit,closesthit,miss,callable,mesh,amplification,"
```

Comment:
Worth noting, the creation of the attribute is right here.

The distinction of decoupling validating and creating the attributes is arbitrary. DXC also does something _really_ silly here where it creates attributes regardless of whether or not the attribute being created is valid. You can see that in [this example](https://godbolt.org/z/7Pa6Pe96s) where the shader attribute is created with an empty string as its shader type.

The correct solution here should be to not create the attribute at all.